src/hooks/useLocalstorage.ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import { useState } from 'react';
+
+export const useLocalstorage
+= <T>(key: string, initialValue: T[]): [T[], (arg: T[]) => void] => {
+  const [value, setValue] = useState<T[]>(() => {
+    try {
+      return JSON.parse(localStorage.getItem(key) || '') || initialValue;
+    } catch {
+      return initialValue;
+    }
+  });
+
+  const save = (value: T[]) => {
+    setValue(value);
+    localStorage.setItem(key, JSON.stringify(value));
+  };
+
+  return [value, save];
```

Comment:
Better practice will be ruturn object, to be sure what you use and no need to remember what index is what
```suggestion
  return {value, save};
```